### ISSUE TITLE
CopyDir/Copy fail on Linux with error code 20

### ISSUE THREAD

**Issue by glek** [fastbuild/fastbuild#155]
I recently started trying to use the `CopyDir` function to copy data folders for my project into the release/debug build locations.

Every time I try I get the following errors.

```
3> Copy set last write time failed (error 20) './release/linux/data/vShader.vert'
4> Copy set last write time failed (error 20) './release/linux/data/fShader.frag'
```

These errors are not specific to these files but seem to occur for any file I attempt to copy.

More information is available on request.

**Comment by ffulin** [comment]
That's quite strange. It implies the copy worked, but then we failed to set the timestamp on the destination file.  The error returned seems to be ENOTDIR.

The only thing that stands out as weird (and perhaps correlates with the ENOTDIR) is that the path in the output string starts with ./

This is off because FASTBuild generally deals with absolute paths when finally dealing with the file system, so I'd actually expect that to be a full path.  Perhaps there is a bug with FASTBuild's path expansion when setting up the CopyDir function while parsing the bff.

If you are able to trim down your bff into a simple snippet for just the copy that reproduces the problem then it would really help narrow down the cause.

**Comment by glek** [comment]
I was a bit mystified when this didn't happen in my test case, but after a quick check I can confirm this bug only happens with version 0.91.

Here is my sample `fbuild.bff`:

```
CopyDir('test') {
    .SourcePaths = "testDir/"
    .Dest = "testDirDest/data/"
}

Alias('all') {
    .Targets = {
        'test'
    }
}
```

The folder `testDir` contains the following file, called `fShader.frag`:

```
#version 300 es
precision mediump float;
uniform sampler2D texture;
in vec2 uv;
out vec4 color;
void main(void) {
    color = texture2D(texture, uv);
}
```

**Comment by ffulin** [comment]
Thanks for the extra info. I'll try to reproduce this.

The CopyNode was modified in v0.91 as part of some ongoing internal re-factoring. It's likely this change in behaviour was introduced then.

**Comment by ffulin** [comment]
I was able to reproduce the issue where the path was not being correctly expanded.  It did not manifest in the exact same failure for me, but I'm fairly confident this is the cause of the issue you were seeing. A fix has been made in the Perforce mainline which I've also integrated into the GitHub "dev" branch (https://github.com/fastbuild/fastbuild/commit/1cbd670bf42255b6e68d07da40307262eea9c7ee). This fix will be in the next release (v0.92).  I've also expanded test coverage to include CopyDir to prevent issues like this in the future.

Thanks for reporting the bug! I'll leave this issue open until the next version is released.
